File: CS/PS/Site/programmers/lv1/체육복.js
function solution(n, lost, reserve) {
  var answer = 0;
  let student = new Array(n + 1).fill(1);
  student[0] = 0;
  
  for(let lostNumber of lost){
    student[lostNumber]--;
  }
  for(let reserveNumber of reserve){
    student[reserveNumber]++;
  }
  for(let i = 1; i <= n; i++){
    for(let lostNumber of lost){
      if (student[lostNumber] == 0){
        if (student[lostNumber - 1] == 2){
          student[lostNumber - 1]--;
          student[lostNumber]++;
          break;
        }
        if (student[lostNumber + 1] == 2){
          student[lostNumber + 1]--;
          student[lostNumber]++;
          break;
        }
      }
    }
  }
  return student.filter((v) => v >=1).length;
}

console.log(solution(4, [2,4],[3]));

File: CS/PS/복습/progr_lv1/체육복.py
# 18:30 ~ #18:38

def solution(n, lost, reserve):
    answer = 0
    #4번은 3, 5번에게만 빌릴 수 있음
    cnt = [0] * (n + 1)
    for l in lost:
        cnt[l] -= 1
    for re in reserve:
        cnt[re] += 1
    for idx in range(1, len(cnt) - 1):
        if cnt[idx] == -1 and cnt[idx - 1] == 1:
            cnt[idx] += 1
            cnt[idx - 1] -= 1
        elif cnt[idx] == -1 and cnt[idx + 1] == 1:
            cnt[idx] += 1
            cnt[idx + 1] -= 1
    for i in range(1, len(cnt)):
        if cnt[i] >= 0:
            answer += 1
    return answer